packages/machine-extractor/src/MachineExtractResult.ts

```diff
@@ -1490,16 +1521,32 @@ export class MachineExtractResult {
             );
           }
           if (!existingLayoutComment) {
-            const calleePosition = {
-              line: this.machineCallResult.callee.loc!.start.line - 1,
-              column: this.machineCallResult.callee.loc!.start.column,
-              index: this.machineCallResult.callee.start!,
+            const definitionNode = this.machineCallResult.definition?.node;
+            if (!t.isObjectExpression(definitionNode)) {
+              const calleePosition = {
+                line: this.machineCallResult.callee.loc!.start.line - 1,
+                column: this.machineCallResult.callee.loc!.start.column,
+                index: this.machineCallResult.callee.start!,
+              } as const;
+
+              layoutEdit = {
+                // this is used as a replace but it could be a simpler~ insertion
+                type: 'replace',
+                range: [calleePosition, calleePosition],
+                newText: `\n/** @xstate-layout ${edit.layoutString} */\n`,
+              };
+              break;
+            }
+            const insidePosition = {
+              line: definitionNode.loc!.start.line - 1,
+              column: definitionNode.loc!.start.column + 1,
+              index: definitionNode.start! + 1,
             } as const;
 
             layoutEdit = {
               // this is used as a replace but it could be a simpler~ insertion
               type: 'replace',
-              range: [calleePosition, calleePosition],
+              range: [insidePosition, insidePosition],
               newText: `\n/** @xstate-layout ${edit.layoutString} */\n`,
```

Comment:
```suggestion
              newText: `\n/** @xstate-layout ${edit.layoutString} */`,
```
nit: I prefer not to have an empty line between the comment and the next thing. But I understand it's subjective, so pick whichever you prefer. I'm only suggesting this here, but if it's changed, I guess it should also be changed in the block above this starting line 1536.

**Indentation**
Right now, we don't ignore the existing indentation when we insert. Is it too naive to use something like this:
```ts
...
const indent = (definitionNode.loc as any).lines.infos[
  definitionNode.loc!.start.line
].indent;
const indentString = ' '.repeat(indent); // not sure this is fully compatible with tabs for indent, but I guess that would be handled by the user's formatter if it is a problem

layoutEdit = {
  // this is used as a replace but it could be a simpler~ insertion
  type: 'replace',
  range: [insidePosition, insidePosition],
  newText: `\n${indentString}/** @xstate-layout ${edit.layoutString} */`,
};
...
```

Reply:
Would this insert an empty line, or would it merely make sure that the machine config code is on the next line?

Reply:
In the test I did it created a new empty line after the layout comment